File: esbmc.cpp/validador_preco/validador_max_duas_casas_decimais.cpp
#define MAX_PRICE_LEN 6

using namespace std;

bool validate_max_two_decimal_places(const char price[MAX_PRICE_LEN], int len) {
    int dot_count = 0;
    int decimals = 0;
    
    for (int i = 0; i < len; i++) {
        if (price[i] == '.') {
            dot_count++;
        }
        if (dot_count == 1 && isdigit(price[i])) {
            decimals++;
        }
    }

    __ESBMC_assert(decimals <= 2, "O preço não pode ter mais de duas casas decimais.");
    return true;
}

int main() {
    char price[MAX_PRICE_LEN];
    int len;

    __ESBMC_assume(len >= 1 && len <= 5);

    validate_max_two_decimal_places(price, len);

    return 0;
}

File: esbmc.py/Testes02/validador_preco/validador_max_duas_casas_decimais.py
MAX_PRICE_LEN: int = 6

def validate_max_two_decimal_places(price: str, length: int) -> bool:
    dot_count: int = 0
    decimals: int = 0
    i: int = 0

    while i < length:
        c: str = price[i]
        if c == '.':
            dot_count += 1
        if dot_count == 1 and c.isdigit():
            decimals += 1
        i += 1

    assert decimals <= 2, "O preço não pode ter mais de duas casas decimais."
    return True


def main() -> None:
    price: str = input()
    length: int = len(price)

    assert 1 <= length <= 5

    validate_max_two_decimal_places(price, length)


main()
